grammar/abnf-grammar.txt

```diff
@@ -1000,10 +1000,9 @@ print-call = print-function print-arguments
 
 ; An annotation consists of an annotation name (which starts with `@`)
 ; with optional annotation arguments, which are identifiers.
-; Note that no parentheses are used if there are no arguments.
 
 annotation = annotation-name
-             [ "(" identifier *( "," identifier ) ")" ]
+             [ "(" *( identifier "," ) [ identifier ] ")" ]
```

Comment:
This change looks fine with me, but I wanted to highlight it, since other things (particularly the ACL2 development) depend on the shared definition of the ABNF grammar.

This lets us have trailing commas in annotation arguments, which is something we wanted to have for all the comma-separated sequences eventually, per https://github.com/AleoHQ/leo/issues/1408#issuecomment-952127123.

This also lets us use `()` for no annotation arguments, e.g. `@test()` as well as `@test`. These are presumably parsed to the same AST.